RefugeeLand.Core.Api.Tests.Unit/Foundations/RefugeeServiceTests.Exceptions.Add.cs

```diff
@@ -0,0 +1,188 @@
+﻿// -------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO DELIVER HUMANITARIAN AID, HOPE AND LOVE
+// -------------------------------------------------------
+
+using EFxceptions.Models.Exceptions;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using RefugeeLand.Core.Api.Models.Refugees;
+using RefugeeLand.Core.Api.Models.Refugees.Exceptions;
+using Xunit;
+
+namespace RefugeeLand.Core.Api.Tests.Unit.Foundations
+{
+    public partial class RefugeeServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowCritcalDependencyExceptionOnAddIfSqlErrorOccursAndLogItAsync()
+        {
+            // given
+            Refugee someRefugee = CreateRandomRefugee();
+            SqlException sqlException = GetSqlException();
+
+            FailedRefugeeStorageException failedRefugeeStorageException =
+                new FailedRefugeeStorageException(sqlException);
+
+            var expectedRefugeeDependencyException =
+                new RefugeeDependencyException(failedRefugeeStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                    broker.GetCurrentDateTimeOffset())
+                .Throws(sqlException);
+
+            // when
+            ValueTask<Refugee> addRefugeeTask =
+                this.refugeeService.AddRefugeeAsync(someRefugee);
+
+            // then
+            await Assert.ThrowsAsync<RefugeeDependencyException>(() =>
+                addRefugeeTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                    broker.LogCritical(It.Is(SameExceptionAs(
+                        expectedRefugeeDependencyException))),
+                Times.Once);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                    broker.GetCurrentDateTimeOffset(),
+                Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                    broker.InsertRefugeeAsync(It.IsAny<Refugee>()),
+                Times.Never);
```

Comment:
The business logic calls `dateTimeBroker` to throw exception, then `loggingBroker` is called. So this should be the order of `.Verify()` statements. The same applies to `VerifyNoOtherCalls()`